common/djangoapps/cors_csrf/helpers.py

```diff
@@ -0,0 +1,92 @@
+"""Helper methods for CORS and CSRF checks. """
```

Comment:
Note to reviewers: This looks like a lot of code, but it's mostly stuff I moved from `cors_csrf.middleware` so it could be re-used by the new `SessionAuthentication` subclass.